<?xml version="1.0" ?>

<launch>
    <arg name="robot_type" default="aliengo" doc="Robot type: [a1, aliengo, go1, laikago]"/>

    <arg name="cheater" default="true"/>

    <!-- make the files into global parameters -->
    <param name="urdfFile" value="/tmp/legged_control/$(arg robot_type).urdf"/>
    <param name="taskFile" value="$(find legged_controllers)/config/$(arg robot_type)/task.info"/>
    <param name="referenceFile" value="$(find legged_controllers)/config/$(arg robot_type)/reference.info"/>
    <param name="gaitCommandFile" value="$(find legged_controllers)/config/$(arg robot_type)/gait.info"/>

    <rosparam file="$(find legged_controllers)/config/controllers.yaml" command="load"/>

    <node if="$(arg cheater)" name="controller_loader" pkg="controller_manager" type="controller_manager"
          output="screen" args="load
          controllers/joint_state_controller
          controllers/legged_controller
          controllers/legged_cheater_controller
"/>

    <node unless="$(arg cheater)" name="controller_loader" pkg="controller_manager" type="controller_manager"
          output="screen" args="load
          controllers/joint_state_controller
          controllers/legged_controller
"/>

    <node pkg="ocs2_legged_robot_ros" type="legged_robot_gait_command" name="legged_robot_gait_command"
          output="screen"/>

    <node pkg="legged_controllers" type="legged_target_trajectories_publisher" name="legged_robot_target"
          output="screen"/>

    <node name="rqt_controller_manager" pkg="rqt_controller_manager" type="rqt_controller_manager" respawn="false" output="screen" />

    <node pkg="tf2_ros" type="static_transform_publisher" name="odom_tf_publisher"
          args="0 0 0 0 0 0 1 map odom"/>
    <node pkg="tf2_ros" type="static_transform_publisher" name="world2map"
          args="0 0 0 0 0 0 1 world map"/>
      <node name="rviz" pkg="rviz" type="rviz" args="-d $(find legged_unitree_description)/launch/rviz.rviz"/>

</launch>
